### ISSUE TITLE
Add dim parameter to torch.bucketize

### ISSUE THREAD

**Issue by Aure20** [pytorch/pytorch#149118]
### 🚀 The feature, motivation and pitch

Currently I need to modify a 2D tensor but I want to use different boundaries for different rows, what I am doing is to use list comprehension and bucketizing each row one by one and then stack the list again (see example). It would be convenient to have a dim parameter and therefore just need to write:
bucketized = torch.bucketize(values, boundaries, dim=1) 
where the boundary tensor shape needs to match the shape along the dim we are operating.
I think to call coniguous just once before should be enough to have a good memory layout on the boundary tensor as well like in the 1D case.

### Alternatives

Example of the current solution I use:
values = torch.tensor([[-0.2, 1.5, 3.2], [2.1, 4.3, 0.5]]) 
boundaries = torch.tensor([[0.0, 1.0, 2.0, 3.0], [2.0, 3.0, 4.0, 5.0]])
bucketized = torch.stack([torch.bucketize(values[i], boundaries[i]) for i in range(values.shape[0])])

**Comment by albanD** [comment]
Hey!

This sounds like a good feature to add.
We would need to also add multiple dimensions to the "binaries" argument but sounds feasible.
We would need a detailed plan here and then someone can implement it !
